packages/taquito/src/contract/rpc-estimate-provider.ts

```diff
@@ -9,29 +10,39 @@ import {
 import {
   DelegateParams,
   isOpWithFee,
+  isOpRequireReveal,
   OriginateParams,
   ParamsWithKind,
   PrepareOperationParams,
   RegisterDelegateParams,
   RPCOperation,
   TransferParams,
+  RevealParams,
 } from '../operations/types';
 import { Estimate } from './estimate';
-import { EstimationProvider } from './interface';
+import { EstimationOptions, EstimationProvider } from './interface';
 import {
   createOriginationOperation,
   createRegisterDelegateOperation,
+  createRevealOperation,
   createSetDelegateOperation,
   createTransferOperation,
 } from './prepare';
-import { Protocols } from '../constants'
 
 interface Limits {
   fee?: number;
   storageLimit?: number;
   gasLimit?: number;
 }
 
+interface EstimateProperties {
+  milligasLimit: number,
+  storageLimit: number,
+  opSize: number,
+  minimalFeePerStorageByteMutez: number,
+  baseFeeMutez?: number
+}
+
```

Comment:
When we do the estimation, the fee is calculated, have a look at the suggestedFeeMutez method of the estimate class. We simulate the operation using the rpc, this gives us the gas that will be consumed by the operation, we use this value in the calculus and the operation size in bytes. I added a constant for the size of the reveal operation as this will not change, but the gas could change over the protocol.
I am not sure I understand your question correctly, don't hesitate to ask again if I don't answer the question.

Reply:
The estimate has a total fee property that includes reveal fee, so this should be okay.